Changes

santoja opened the pull request:
Hello @infojunkie! 
I had changed a lot of things on the code! 
Now it's work don't mather where the MantisBT is installed.
Changes:
- The name is now just Lightbox, so the user don't need to change the folder name when download it
- Last version of Lightbox
- Use the plugin_file for the Lightbox files and lightbox.js file
- Location of the files to the folder "files"
- Hardcode change on the image location on lightbox.css. Work, but it's ugly

santoja committed:
New version. Less hardcode, more integrate.

santoja committed:
Removed original Lightbox.js

santoja committed:
Update Lightbox.php

License update

dregad commented on Lightbox.php:
```diff
@@ -22,7 +22,7 @@ class LightboxPlugin extends MantisPlugin {
+        $this->version = '0.1.1';
         $this->requires = array(
             'MantisCore' => '>= 1.2.0',
         );
```
With this requirement, the plugin will be disabled on Mantis versions >= 1.3, therefore the code below is not useful.

This is how the API works:
If the plugin's minimum dependency for MantisCore is unspecified or lower than the current release (i.e. the plugin does not specifically list the current core version as supported) and the plugin does not define a maximum dependency, we add one with the current version's minor release (i.e. for 1.3.1 we would add '<1.3').

The purpose of this is to avoid compatibility issues by disabling plugins which have not been updated for a new Mantis release; authors have to revise their code (if necessary), and release a new version of the plugin with updated dependencies.

To indicate compatibility (e.g. with release 1.3), one can either:
1. update the minimum required version (i.e. plugin only works with 1.3; 1.2-compatible code is maintained separately):
   `$this->requires = array( 'MantisCore' => '1.3' );`
2. add the new release as a 2nd minimum version (i.e. the plugin is  compatible with both versions):
   `$this->requires = array( 'MantisCore' => '1.2, 1.3' );`
3. add a maximum version higher than the new release:
   `$this->requires = array( 'MantisCore' => '1.2, <2.0' );`
   Note that this may cause the plugin to face compatibility issues   if and when a version 1.4 is released.

santoja replied:
Thanks @dregad!
I'll change the requirement to 1.2 and 1.3 version.

dregad commented:
With regards to hosting this under mantisbt-plugins organization, please let me know how you'd like to handle this in terms of who the maintainer(s) will be and what the, then we can work out the details (i.e. the source for the fork, access rights, etc).

santoja commented:
@dregad I don't know if @infojunkie still active, but you can add me and him for maintenance purpose.

santoja committed:
Now it's compatible with MantisBT 1.2 and 1.3

Version requires changed. Thanks @dregad

santoja committed:
Update Lightbox.php

dregad commented:
I suggest to wait a few days to give @infojunkie a chance to respond.

santoja committed:
Update Lightbox.php

infojunkie commented:
Thanks for the pull request - I will review and comment shortly.

santoja committed:
Version 0.2

Version 0.2 released!
Added config for the plugin: Now user can choose where he wants a
lightbox
New english and portuguese strings
Minor changes on README.md

santoja commented:
Hello @infojunkie.
I coded a config page to the plugin.

santoja commented on Lightbox.php:
```diff
@@ -17,22 +18,33 @@
-        $this->version = '0.1';
+        $this->name = plugin_lang_get('title');
+        $this->description = plugin_lang_get('description');
+        $this->version = '0.2';
```
Version change, alot of new things

santoja commented on Lightbox.php:
```diff
@@ -17,22 +18,33 @@
+        $this->page = 'config';
         $this->requires = array(
-            'MantisCore' => '>= 1.2.0',
+            'MantisCore' => '>=1.2, <1.4',
```
Working on 1.2 and 1.3

santoja commented on Lightbox.php:
```diff
@@ -17,22 +18,33 @@
+    /**
+     * Default plugin configuration.
+     */
+    function config() {
```
Default config for plugin

santoja commented on Lightbox.php:
```diff
@@ -41,13 +53,17 @@ function hooks() {
+        if (end($currentUrl) !== 'view.php') return;
+        
+        return '        <script type="text/javascript">'.
+                        'var lightbox_display_on_img_preview = '.plugin_config_get('display_on_img_preview').';'.
```
Using the plugin_config_get and plugin_file for config variables and file location

santoja committed:
Add "choose the extentions"

New option add to let you choose the extensions affected by the lightbox

santoja commented:
@infojunkie @dregad Can we put it on Mantis-BT official Github? They can give us access to de repository and that stuff.

dregad commented:
As mentioned before, I have no issues with having this under mantisbt-plugins org, but would like to have @infojunkie's agreement to transfer ownership of the repo.

infojunkie commented:
I can add you as collaborator here, @santoja. Then you or I can publish the plugin to `mantisbt-plugins`. This would save unneeded forks. I just prefer to keep the name `MantisBT-Lightbox`, because `Lightbox` on its own is misleading. 

Would that work for you?

santoja commented:
Great @infojunkie! Thank you! =D

infojunkie commented:
OK done - feel free to proceed with merging your own pull request.

dregad commented:
>  I just prefer to keep the name MantisBT-Lightbox, because Lightbox on its own is misleading. 

I understand that, but if you look at this from our perspective, we just can't have every other plugin be named "MantisBT-XXX' - it is obvious if you're a part of _mantisbt-plugins_ organization that this is about MantisBT, without having it repeated in the plugin name.

You're free to keep the current name and not have your plugin in the org. Your call.

infojunkie commented:
Oh I meant on this repo, not the mantisbt-plugins mirror. I've no problem renaming it there :-)
